### PR TITLE
[SPARK-43359][SQL] Delete from Hive table should throw "UNSUPPORTED_FEATURE.TABLE_OPERATION"

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to fix error exception about 'DELETE from Hive table'

### Why are the changes needed?
Proper names of error classes should improve user experience with Spark SQL.

**BEFORE**
```
scala> sql("delete from t")
org.apache.spark.SparkException: [INTERNAL_ERROR] Unexpected table relation: HiveTableRelation [`spark_catalog`.`default`.`t`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#0], Partition Cols: []]
```

**AFTER**
```
scala> sql("delete from t")
org.apache.spark.sql.AnalysisException: [UNSUPPORTED_FEATURE.TABLE_OPERATION] The feature is not supported: Table `spark_catalog`.`default`.`t` does not support DELETE. Please check the current catalog and namespace to make sure the qualified table name is expected, and also check the catalog implementation which is configured by "spark.sql.catalog".
```

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA & Add new UT.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @srielau @MaxGekk

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala [hunk 1]
```diff
@@ -3098,4 +3098,20 @@ class HiveDDLSuite
       "CREATE TABLE tab (c1 int) PARTITIONED BY (c1) STORED AS PARQUET",
       "Cannot use all columns for partition columns")
   }
+
+  test("SPARK-43359: Delete table not allowed") {
+    val tbl = "T1"
+    withTable(tbl) {
```
Comment (MaxGekk) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala:3104]: Where is `tbl` used besides of `withTable `? Could you use it in the sql statements below, please.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala (hunk 1 above)
Comment (panbingkun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala:3104]: I have already fixed it

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala [hunk 2]
```diff
@@ -3098,4 +3098,20 @@ class HiveDDLSuite
...
+      sql(s"CREATE TABLE $tbl(c1 INT)")
+      val e = intercept[AnalysisException] {
+        sql(s"DELETE FROM $tbl WHERE c1 = 1")
+      }
+      checkError(e,
+        errorClass = "UNSUPPORTED_FEATURE.TABLE_OPERATION",
+        parameters = Map(
+          "tableName" -> "`spark_catalog`.`default`.`t1`",
```
Comment (MaxGekk) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala:3112]: How about this one?

Comment (panbingkun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: Ok, chang to s"`spark_catalog`.`default`.`$tbl`"

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @panbingkun and @MaxGekk .
I revised the PR description by adding the **BEFORE** and **AFTER** error message.

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 3.5.0.

**Pull request closed without merging**